html_app/ui/microscopy.soy

```diff
@@ -308,110 +308,28 @@
-					 <input class="scb_f_microscopy_select_conditions" microscopy_id='{$microscopy.id}'
-                       assignment_id='{$assignment.id}' experiment_id='{$experiment.id}' type="radio" value="HnE" cell_treatment_id='{$r.cell_treatment.id}'
-                       lane_kind="{$r.kind}" lane_id="{if $r.kind == 'existing'}{$r.lane.id}{/if}" 
-						{if $r.is_sample_enabled and length(keys($assignment.template.slide_parser[$r.lane.cell_treatment.treatment_list.first.collection_id][$r.lane.kind])) == 1}checked="checked"{/if} /><span class="scb_s_western_blot_choose_gel_type_input_text ">{$assignment.template.micro_kinds.Dye.conditions.HnE.name}</span>
-					{/if}
-					</div>
-					{/foreach}
-				{/if}
-			{elseif $assignment.id =='assignment_706_2014' and not ($r.kind == 'placeholder')}	
-				<div>
-					 <input class="scb_f_microscopy_select_conditions {if $r.is_sample_enabled}{else}scb_s_microscopy_dim_conditions{/if}" microscopy_id='{$microscopy.id}'
-                       assignment_id='{$assignment.id}' experiment_id='{$experiment.id}' type="radio" value="rgb" cell_treatment_id='{$r.cell_treatment.id}'
-                       lane_kind="{$r.kind}" lane_id="{if $r.kind == 'existing'}{$r.lane.id}{/if}" 
-						{if $r.is_sample_enabled and length(keys($assignment.template.slide_parser[$r.lane.cell_treatment.treatment_list.first.collection_id][$r.lane.kind])) == 1}checked="checked"{/if} /><span class="scb_s_western_blot_choose_gel_type_input_text scb_s_microscopy_dim_conditions">{$assignment.template.micro_kinds.IF.conditions.rgb.name}</span>
-				</div>
+            {if $r.lane}
+                {foreach $condition in $r.lane.cell_treatment.treatment_list.first.conditions[$r.lane.kind]}
+                    <!--If there is only one condition, radio button is already checked-->
+                    <div>
+                        <input class="scb_f_microscopy_select_conditions {if $r.is_sample_enabled}{else}scb_s_microscopy_dim_conditions{/if}" microscopy_id='{$microscopy.id}'
+                        assignment_id='{$assignment.id}' experiment_id='{$experiment.id}' type="radio" value="{$condition}" cell_treatment_id='{$r.cell_treatment.id}'
+                        lane_kind="{$r.kind}" lane_id="{if $r.kind == 'existing'}{$r.lane.id}{/if}"
+                        {if $r.is_sample_enabled and ($r.lane.slide_conditions == $condition or length($r.lane.cell_treatment.treatment_list.first.conditions[$r.lane.kind]) == 1)}checked="checked"{/if} />
+                        <span class="scb_s_western_blot_choose_gel_type_input_text {if $r.is_sample_enabled}{else}scb_s_microscopy_dim_conditions{/if}">
+                        {$assignment.template.micro_kinds[$r.lane.kind].conditions[$condition].name}</span>
+                    </div>
+                {/foreach}
 			{/if}
+
             </td>
+            <!--Remove bucket-->
 			<td class='scb_s_microscopy_samples_table_td {if $r.display_sample}{else}scb_s_experiment_setup_table_border{/if}'>
```

Comment:
again, spacing is weird here, we have some tabs and some spaces, it would be nice to be consistent....